tarantool_test.go

```diff
@@ -385,6 +385,81 @@ func BenchmarkClientParallelMassiveUntyped(b *testing.B) {
 	close(limit)
 }
 
+func BenchmarkConnection_PointSelect(b *testing.B) {
+	b.StopTimer()
+	conn, err := Connect(server, opts)
+	if err != nil {
+		b.Errorf("No connection available")
+		return
+	}
+	defer conn.Close()
+
+	schema := conn.Schema
+	rSpaceNo, rIndexNo, err := schema.ResolveSpaceIndex("test_perf", "primary")
+	if err != nil {
+		b.Errorf("symbolic space and index params not resolved")
+	}
+	b.StartTimer()
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			_, err := conn.Select(rSpaceNo, rIndexNo, 0, 1, IterEq, []interface{}{uint32(1)})
+			if err != nil {
+				b.Errorf("No connection available: %s", err)
+				return
+			}
+		}
+	})
+}
```

Comment:
Isn't it the same as `BenchmarkClientParallel()`? If so, let's just leave it and don't add the benchmark with the same meaning.

TBH, I didn't know that there are a bunch of benchmarks already, when writting the issue. Sorry if it confuses you.

----

While looking into the existing benchmarks I wrote several schemas to understand what is what. I placed them below. Feel free to use to make things more clear.

```
serial (in one goroutine)

        N
 +-------------+
 |             |
 v             v
 [ ][ ][ ][ ][ ]
 
future (in one goroutine)

  500 in a line
 +-------------+
 |             |
 v             v
 [ ][ ][ ][ ][ ] <-+
      wait         |
 [ ][ ][ ][ ][ ]   | ceil(N/500)
      wait         | 
 [ ][ ][ ][ ][ ]   | 
      wait       <-+ 
      
future parallel (in goroutines amount same as CPUs count)

                               CPUs count
       +-----------------+-----------------+-----------------+
       |                 |                 |                 |
       v                 v                 v                 v
  500 in a line     500 in a line     500 in a line     500 in a line
 +-------------+   +-------------+   +-------------+   +-------------+
 |             |   |             |   |             |   |             |
 v             v   v             v   v             v   v             v
 [ ][ ][ ][ ][ ]   [ ][ ][ ][ ][ ]   [ ][ ][ ][ ][ ]   [ ][ ][ ][ ][ ]
      wait              wait              wait              wait
 [ ][ ][ ][ ][ ]   [ ][ ][ ][ ][ ]   [ ][ ][ ][ ][ ]   [ ][ ][ ][ ][ ]
      wait              wait              wait              wait
 [ ][ ][ ]         [ ][ ][ ]         [ ][ ][ ]         [ ][ ][ ]
      wait              wait              wait              wait
      
N squares at sum in all goroutines.

parallel (in goroutines amount same as CPUs count)

                             CPUs count
        +----------------+----------------+----------------+
        |                |                |                |
        v                v                v                v
     ~N/CPUs          ~N/CPUs          ~N/CPUs          ~N/CPUs
 +-------------+  +-------------+  +-------------+  +-------------+
 |             |  |             |  |             |  |             |
 v             v  v             v  v             v  v             v
 [ ][ ][ ][ ][ ]  [ ][ ][ ][ ][ ]  [ ][ ][ ][ ][ ]  [ ][ ][ ][ ][ ]
 
N squares at sum in all goroutines.
```

Here `N` is `b.N`, `500` is `N` (I would rename it to `BATCH_SIZE`).

I have no a diagram for 'massive' benchmarks, because it is not clear for me, what they're doing.

Reply:
Benchmark with the same meaning removed.